tdms/tests/system/data/input_generation/matfile_option_edit.py

```diff
@@ -0,0 +1,99 @@
+import os
+from dataclasses import dataclass
+from pathlib import Path
+
+import numpy as np
+from hdf5storage import loadmat, savemat
+
+LOCATION_OF_THIS_FILE = os.path.abspath(os.path.dirname(__file__))
+
+
+@dataclass
+class OverwriteWith:
+    """Small container class that will be used when overwriting the "flag" variables usecd and intmethod when a config file specifies an 'adjust' field.
+
+    Intended workflow along the lines of:
+    intmethod = OverwriteWith(whether_to_overwrite, value_to_replace_with)
+    if intmethod.write:
+        h5data['intmethod'] = intmethod.value
+    else:
+        pass
+    """
+
+    # Flags whether or not to overwrite a value with self.value
+    write: bool
```

Comment:
```suggestion
    to_overwrite: bool
```
(Although this suggestion should be done via VSCode's rename symbol, if you like it)

Reply:
I was thinking of the ultimate logic...
```python
if illmethod.should_be_overwritten:
    # flibble
```